README.md

```diff
@@ -1,8 +1,293 @@
-# ipc-agent
-Implementation of an IPC agent
+# IPC Agent
+> The InterPlanetary Consensus (IPC) orchestrator.
+
+The IPC Agent is the entrypoint to interacting with IPC. It is a client application that provides a simple and easy-to-use interface to interact with IPC as a user and run all the processes required for the operation of a subnet.
 
 See the [docs](docs) for a conceptual overview.
+a
+## Installation
+### Build requirements
+To build the IPC Agent you need to have Rust installed in your environment. The current MSRV (Minimum Supported Rust Version) is nightly-2022-10-03 due to some test build dependencies. A working version is tracked in rust-toolchain (this is picked up by rustup automatically). You can look for instructions on [how to run Rust and rustup following this link](https://www.rust-lang.org/tools/install).
+
+> According to the operating system you are running, you may have to install additional dependencies not installed in your system to follow these instructions like `build-essentials`, `libssl-dev`, `git`, `curl`. If something fails while building the binaries double-check these dependencies.
+
+### Build instructions
+To build the binary for the IPC agent you need to build the requirements in your environment, clone this repo, and build the binary following these steps:
+```bash
+$ git clone https://github.com/consensus-shipyard/ipc-agent.git
+$ cd ipc-agent
+$ rustup target add wasm32-unknown-unknown
+$ make build
+```
+
+This builds the binary of the IPC agent in the `./bin` folder of your repo. If you want to make the command available everywhere, add this folder to the binary `PATH` of your system. To see if the installation was successfully you can run the following command: 
+``` bash
+$ ./bin/ipc_agent --help
```

Comment:
I just realised I'm very tripped by this even after running trough several times. Unix convention is for commands to use `-` instead of `_`. Think `apt-get`, `linux-version`, `pkg-config`, `ssh-keygen`, ... Is there a reason why we went with underscore?

Reply:
Amazing, question. Rust uses `_` for packages, I think we inherited it that from the crate name. But I agree that we should probably use `-`. Let me start a quick poll in Slack just in case (before I do the whole renaming).